3203-1_1.xml

```diff
@@ -0,0 +1,128 @@
+POSSIBILITY OF SUCH DAMAGE.
+-->
+
+<LWM2M  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="http://openmobilealliance.org/tech/profiles/LWM2M.xsd">
+	<Object ObjectType="MODefinition">
+		<Name>Analog Output</Name>
+		<Description1>This IPSO object is a generic object that can be used with any kind of analog output interface.</Description1>
+		<ObjectID>3203</ObjectID>
+		<ObjectURN>urn:oma:lwm2m:ext:3203:1.1</ObjectURN>
+		<LWM2MVersion>1.0</LWM2MVersion>
+		<ObjectVersion>1.1</ObjectVersion>
+		<MultipleInstances>Multiple</MultipleInstances>
+		<Mandatory>Optional</Mandatory>
+		<Resources>
+			<Item ID="5650">
+				<Name>Analog Output Current Value</Name>
+				<Operations>RW</Operations>
+				<MultipleInstances>Single</MultipleInstances>
+				<Mandatory>Mandatory</Mandatory>
+				<Type>Float</Type>
+				<RangeEnumeration>0..1</RangeEnumeration>
+				<Units></Units>
+				<Description>The current state of the analogue output.</Description>
+			</Item>
+			<Item ID="5750">
+				<Name>Application Type</Name>
+				<Operations>RW</Operations>
+				<MultipleInstances>Single</MultipleInstances>
+				<Mandatory>Optional</Mandatory>
+				<Type>String</Type>
+				<RangeEnumeration></RangeEnumeration>
+				<Units></Units>
+				<Description>If present, the application type of the actuator as a string, for instance, "Valve"</Description>
+			</Item>
+			<Item ID="5603">
+				<Name>Min Range Value</Name>
+				<Operations>R</Operations>
+				<MultipleInstances>Single</MultipleInstances>
+				<Mandatory>Optional</Mandatory>
+				<Type>Float</Type>
+				<RangeEnumeration></RangeEnumeration>
+				<Units></Units>
+				<Description>The minimum value that can be set for the output</Description>
+			</Item>
+			<Item ID="5604">
+				<Name>Max Range Value</Name>
+				<Operations>R</Operations>
+				<MultipleInstances>Single</MultipleInstances>
+				<Mandatory>Optional</Mandatory>
+				<Type>Float</Type>
+				<RangeEnumeration></RangeEnumeration>
+				<Units></Units>
+				<Description>The maximum value that can be set for the output</Description>
+			</Item>
+			<Item ID="5518">
+				<Name>Timestamp</Name>
+				<Operations>R</Operations>
+				<MultipleInstances>Single</MultipleInstances>
+				<Mandatory>Optional</Mandatory>
+				<Type>Time</Type>
+				<RangeEnumeration></RangeEnumeration>
+				<Units></Units>
+				<Description>The timestamp of when the measurement was performed.</Description>
+			</Item>
+			<Item ID="6050">
+				<Name>Fractional Timestamp</Name>
+				<Operations>R</Operations>
+				<MultipleInstances>Single</MultipleInstances>
+				<Mandatory>Optional</Mandatory>
+				<Type>Float</Type>
+				<RangeEnumeration>0..1</RangeEnumeration>
+				<Units>s</Units>
+				<Description>Fractional part of the timestamp when sub-second precision is used (e.g., 0.23 for 230 ms).</Description>
+			</Item>
+			<Item ID="6042">
+				<Name>Measurement Quality Indicator</Name>
```

Comment:
Also measurement quality for actuators sounds a bit odd.

Reply:
Object 3203 is Analog Output. My reason for adding Measurement Quality is that the output voltage requires a measurement that involves a level of uncertainty that can be reported. I admit that this was something that I hesitated to add, but did it anyway.

Btw, the resource 5650 Analog Output Current Value has likely a bug in range (0..1). The counterpart (Analog Inpout) resource 5600 Analog Input Current Value does not have range limitation.

Reply:
I think the functionality of the value resource is not such that it would be measured, but rather that you can read exactly the value that was written there. Therefore there’s no uncertainty.

Reply:
> I think the functionality of the value resource is not such that it would be measured, but rather that you can read exactly the value that was written there. Therefore there’s no uncertainty.

I'm ready to let this go and leave quality only in sensor objects. After all quality on sensor measurements is what matters to us.

Reply:
Yep, pushed in a change that removed quality resources from Digital Output (3203) and Positioner (3337).

Reply:
Thanks! And just to clarify: if you think there's a valid use case for the uncertainty here, I'm not against keeping them. Right now, with the given semantics of the objects, I just don't see one.